### ISSUE TITLE
[release/2.2.1xx] Test failure during build: extra file "MSBuildTestApp.exe" found after publish

### ISSUE THREAD

**Issue by dagood** [dotnet/cli#9843]
These CLI builds failed building d7b9504c9fd5e4f261cc91a62a449daace623d69 across platforms, build number 20180815.1:

* Linux x64: https://devdiv.visualstudio.com/DefaultCollection/DevDiv/_build/results?buildId=1949917&view=logs
* OSX x64: https://devdiv.visualstudio.com/DefaultCollection/DevDiv/_build/results?buildId=1949920&view=logs
* RHEL6 x64: https://devdiv.visualstudio.com/DefaultCollection/DevDiv/_build/results?buildId=1949921&view=logs
* Windows x86: https://devdiv.visualstudio.com/DefaultCollection/DevDiv/_build/results?buildId=1949923&view=logs

Example failure from the win x86 leg. They all have `MSBuildTestApp.exe` listed as an extra file.

```
DOTNETTEST : error Message:  [E:\A\_work\4262\s\build\test\RunTest.proj]
Following extra files are found inside directory E:\A\_work\4262\s\bin\3\win-x86\test\dotnet-publish.Tests\PublishApp_win7-x86_--self-contained=false\MSBuildTestApp\bin\Release\netcoreapp2.2\win7-x86\publish 
MSBuildTestApp.exe
```

I broke this issue out from the automatically filed https://github.com/dotnet/core-eng/issues/4058 and put it here because it doesn't look like an infra issue. I don't have any context about what this build was for.

/cc @mmitche @livarcocc @wli3

**Comment by peterhuene** [comment]
That's the case.  I'll get the test updated.

**Comment by peterhuene** [comment]
Fixed with #9847.
